File: Sample/HomeApplication/app/src/main/java/m/k/android/sample/homeapplication/MainActivity.java
package m.k.android.sample.homeapplication;

import android.appwidget.AppWidgetManager;
import android.appwidget.AppWidgetProviderInfo;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.content.pm.ResolveInfo;
import android.support.annotation.NonNull;
import android.support.v7.app.ActionBarActivity;
import android.os.Bundle;
import android.util.Log;

import java.util.List;

public class MainActivity extends ActionBarActivity {

    private static final String TAG = "HomeApplication";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final PackageManager packageManager = getPackageManager();
        List<ResolveInfo> applicationInfoList = getApplicationList(packageManager);
        outputApplicationInfoLog(applicationInfoList, packageManager);

        final AppWidgetManager widgetManager = AppWidgetManager.getInstance(this);
        List<AppWidgetProviderInfo> widgetInfoList = getWidgetList(widgetManager);
        outputWidgetInfoLog(widgetInfoList, packageManager);
    }

    @Override
    public void onBackPressed() {
        // disable back key.
    }

    private List<ResolveInfo> getApplicationList(@NonNull PackageManager manager) {
        final Intent intent = new Intent(Intent.ACTION_MAIN);
        intent.addCategory(Intent.CATEGORY_LAUNCHER);
        // see PackageManager.MATCH_DEFAULT_ONLY
        return manager.queryIntentActivities(intent, 0);
    }

    private List<AppWidgetProviderInfo> getWidgetList(@NonNull AppWidgetManager manager) {
        return manager.getInstalledProviders();
    }

    private void outputApplicationInfoLog(List<ResolveInfo> applicationInfoList, @NonNull PackageManager manager) {
        if(applicationInfoList != null) {
            for(ResolveInfo info : applicationInfoList) {
                outputDebugLog("Application : " + info.loadLabel(manager));
            }
        }
    }

    private void outputWidgetInfoLog(List<AppWidgetProviderInfo> widgetInfoList, @NonNull PackageManager manager) {
        if(widgetInfoList != null) {
            for(AppWidgetProviderInfo info : widgetInfoList) {
                outputDebugLog("Widget : " + info.label);
            }
        }
    }

    private void outputDebugLog(String msg) {
        Log.d(TAG, msg);
    }
}

File: LibrarySample/KoinSample/scope/src/main/java/com/github/mag0716/scope/MainActivity.kt
package com.github.mag0716.scope

import android.content.Intent
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import org.koin.android.ext.android.inject
import org.koin.androidx.scope.ext.android.bindScope
import org.koin.androidx.scope.ext.android.getOrCreateScope

class MainActivity : AppCompatActivity() {

    companion object {
        const val TAG = "Scope"
    }

    private val presenter: Presenter by inject()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val scope = getOrCreateScope("Scope")
        bindScope(scope)

        Log.d(TAG, "${presenter.requestHello()}($presenter)")

        button.setOnClickListener {
            startActivity(Intent(this, MainActivity::class.java))
        }
    }
}
